src/components/UI/MainButton.vue

```diff
@@ -1,5 +1,7 @@
 <template>
-  <button :disabled="disabled" :class="buttonClasses"><slot /></button>
+  <button :disabled="disabled" :class="buttonClasses" @click="callback">
```

Comment:
The Vue way of handling communication FROM child component TO parent is by using `events` instead of passing functions as props. There are a couple of advantages and probably the biggest one is encapsulation: child component does not need to carry about if and how parent component handles this event - especially child component don't need to check if the function is passed and if e.g. fails with error.

this is a bit different than in react - more similar to pure HTML/JS

Reply:
I thank you so much for correcting my ways 😃 

I really hope that this is what you meant. I have made a MainButton emit "clicked" event that allows me to trigger action/function in parent component.